src/tsickle.ts

```diff
@@ -105,89 +105,177 @@ class ClosureRewriter extends Rewriter {
-    let sig = typeChecker.getSignatureFromDeclaration(fnDecl);
-
-    // Construct the JSDoc comment by reading the existing JSDoc, if
-    // any, and merging it with the known types of the function
-    // parameters and return type.
-    let jsDoc = this.getJSDoc(fnDecl) || [];
+    let minCount = 9999;
+    let maxCount = 0;
     let newDoc = extraTags;
-
-    // Copy all the tags other than @param/@return into the new
-    // comment without any change; @param/@return are handled later.
-    for (let tag of jsDoc) {
-      if (tag.tagName === 'param' || tag.tagName === 'return') continue;
-      newDoc.push(tag);
+    let paramNamesToReturn: string[] = [];
+    const abstract = {tagName: 'abstract'};
+    const isOverloaded = fnDecls.length > 1;
+    const isConstructor = fnDecls[0].kind === ts.SyntaxKind.Constructor;
+
+    for (let fnDecl of fnDecls) {
+      if (fnDecl.parameters.length < minCount) {
+        minCount = fnDecl.parameters.length;
+      }
+      if (fnDecl.parameters.length > maxCount) {
+        maxCount = fnDecl.parameters.length;
+      }
     }
+    let paramData = {
+      names: new Array<Array<string>>(maxCount),
+      types: new Array<Array<string>>(maxCount),
+      returns: ''
+    };
 
-    // Abstract
-    if ((fnDecl.flags & ts.NodeFlags.Abstract)) {
-      newDoc.push({tagName: 'abstract'});
-    }
+    for (let fnDecl of fnDecls) {
+      // Construct the JSDoc comment by reading the existing JSDoc, if
+      // any, and merging it with the known types of the function
+      // parameters and return type.
+      let jsDoc = this.getJSDoc(fnDecl) || [];
+
+      // Copy all the tags other than @param/@return into the new
+      // comment without any change; @param/@return are handled later.
+      for (let tag of jsDoc) {
+        if (tag.tagName === 'param' || tag.tagName === 'return' ||
+            newDoc.indexOf(tag) !== -1) continue;
+        newDoc.push(tag);
+      }
 
-    // Parameters.
-    if (sig.parameters.length) {
-      // Iterate through both the AST parameter list and the type's parameter
-      // list, as some information is only available in the former.
-      for (let i = 0; i < sig.parameters.length; i++) {
-        let paramNode = fnDecl.parameters[i];
-        let paramSym = sig.parameters[i];
-        let type = typeChecker.getTypeOfSymbolAtLocation(paramSym, fnDecl);
+      // Abstract
+      if ((fnDecl.flags & ts.NodeFlags.Abstract) && (newDoc.indexOf(abstract) === -1)) {
+        newDoc.push(abstract);
+      }
+      // Merge the parameters into a single list of merged names and list of types
+      let sig = typeChecker.getSignatureFromDeclaration(fnDecl);
+      // Parameters.
+      if (sig.parameters.length) {
+        // Iterate through both the AST parameter list and the type's parameter
+        // list, as some information is only available in the former.
+        for (let i = 0, l = sig.parameters.length; i < l; i++) {
```

Comment:
Is the "l = sig.parameters.length" thing for performance?  If so, I'd prefer to leave it to profiling to optimize.